esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -3271,62 +3191,33 @@ bool CFileSprayEx::onOpenSave(IEspContext &context, IEspOpenSaveRequest &req, IE
     return true;
 }
 
-bool CFileSprayEx::getDropZoneFiles(IEspContext &context, const char* dropZone, const char* netaddr, const char* path,
-                                    IEspDropZoneFilesRequest &req, IEspDropZoneFilesResponse &resp)
+void CFileSprayEx::getPhysicalFiles(IEspContext &context, const char *dropZoneName, const char *host, const char *path, const char *fileNameMask, bool directoryOnly, IArrayOf<IConstPhysicalFileStruct> &files)
 {
-    if (!checkDropZoneIPAndPath(context.getClientVersion(), dropZone, netaddr, path))
-        throw MakeStringException(ECLWATCH_DROP_ZONE_NOT_FOUND, "Dropzone is not found in the environment settings.");
-
-    bool directoryOnly = req.getDirectoryOnly();
-
+    SocketEndpoint ep(host);
     RemoteFilename rfn;
-    SocketEndpoint ep;
-#ifdef MACHINE_IP
-    ep.set(MACHINE_IP);
-#else
-    ep.set(netaddr);
-    if (ep.isNull())
-        throw MakeStringException(ECLWATCH_INVALID_INPUT, "CFileSprayEx::getDropZoneFiles: cannot resolve network IP from %s.", netaddr);
-#endif
-
     rfn.setPath(ep, path);
     Owned<IFile> f = createIFile(rfn);
-    if(f->isDirectory()!=fileBool::foundYes)
-        throw MakeStringException(ECLWATCH_INVALID_DIRECTORY, "%s is not a directory.", path);
+    if (f->isDirectory()!=fileBool::foundYes)
+        throw makeStringExceptionV(ECLWATCH_INVALID_DIRECTORY, "%s is not a directory.", path);
 
-    IArrayOf<IEspPhysicalFileStruct> files;
-    Owned<IDirectoryIterator> di = f->directoryFiles(NULL, false, true);
-    if(di.get() != NULL)
+    Owned<IDirectoryIterator> di = f->directoryFiles(nullptr, false, true);
+    ForEach(*di)
     {
-        ForEach(*di)
-        {
-            StringBuffer fname;
-            di->getName(fname);
-
-            if (fname.length() == 0 || (directoryOnly && !di->isDir()))
-                continue;
+        StringBuffer fileName;
+        di->getName(fileName);
 
-            Owned<IEspPhysicalFileStruct> onefile = createPhysicalFileStruct();
+        if ((directoryOnly && !di->isDir()) || (!di->isDir() && !isEmptyString(fileNameMask) && !WildMatch(fileName.str(), fileNameMask, true)))
```

Comment:
Split onto two lines to make it easier to read.  and/or extract the if (di->isDir()) condition out to 
if

Reply:
Ok, I will extract the if (di->isDir()) condition out to.